ios/engine/KMEI/KeymanEngine/Classes/KeyboardHeightViewController.swift

```diff
@@ -0,0 +1,440 @@
+/*
+ * Keyman is copyright (C) SIL International. MIT License.
+ * 
+ * Created by Shawn Schantz on 2024-10-25.
+ * 
+ * View controller for adjusting the height of the keyboard
+ */
+
+import Foundation
+import UIKit
+import os.log
+
+class KeyboardHeightViewController: UIViewController {
+  let contentView: UIView = UIView()
+  let defaultButton = UIButton(type: .roundedRect)
+  let label: UILabel = UILabel()
+  let keyboardImage: UIImageView = UIImageView()
+  let keyboardResizer: UIImageView = UIImageView()
+  var keyboardConstraintsArray: [NSLayoutConstraint] = []
+  var resizerConstraintsArray: [NSLayoutConstraint] = []
+  var isPortrait = true
+  var keyboardHeight = 0.0
+  var defaultPortraitHeight = 0.0
+  var defaultLandscapeHeight = 0.0
+  var minKeyboardHeight = 10.0
+  var maxKeyboardHeight = 10.0
+
+  public init() {
+    super.init(nibName: nil, bundle: nil)
+    _ = view
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    os_log("KeyboardHeightViewController viewDidLoad", log:KeymanEngineLogger.ui, type: .info)
+
+    self.determineOrientation()
+    self.determineDefaultKeyboardHeights()
+    self.applyKeyboardHeight()
+    
+    title = NSLocalizedString("adjust-keyboard-height-title", bundle: engineBundle, comment: "")
+    navigationItem.setHidesBackButton(false, animated: true)
+    navigationItem.leftBarButtonItem?.isEnabled = true
+
+    navigationController?.toolbar?.barTintColor = UIColor.orange
+
+    self.configureContentView()
+    self.configureDefaultHeightButton()
+    self.configureLabel()
+    self.configureKeyboardImage()
+    self.updateKeyboardImage()
+    self.configureKeyboardConstraints()
+    self.configureKeyboardResizer()
+  }
+  
+  override func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
+ 
+    self.calculateKeyboardHeightLimits()
+  }
+  
+  private func determineOrientation() {
+    self.isPortrait = UIScreen.main.bounds.height > UIScreen.main.bounds.width
+    let message = "determineOrientation, isPortrait: \(self.isPortrait)"
+    os_log("%{public}s", log:KeymanEngineLogger.ui, type: .debug, message)
+  }
+  
+  private func determineDefaultKeyboardHeights() {
+    // if no KeyboardScaleMap found for device, then default to 216.0
+    let portraitKeyboardScale = KeyboardScaleMap.getDeviceDefaultKeyboardScale(forPortrait: true)
+    self.defaultPortraitHeight = Double(portraitKeyboardScale?.keyboardHeight ?? 216.0)
```

Comment:
216.0 appears in a few places. Is it worth making it a constant somewhere?

Reply:
I created a new constant `unknownDeviceKeyboardHeight`